kin-recovery/src/main/java/kin/recovery/BackupManager.java

```diff
@@ -48,19 +48,14 @@ public void registerBackupCallback(@NonNull final BackupCallback backupCallback)
 //		this.callbackManager.setBackupEvents(backupEvents);
 //	}
 
-	public void registerRestoreCallback(@NonNull final RestoreCallback restoreCallback) {
+	public void registerRestoreCallback(@NonNull final RestoreCallback<KinAccount> restoreCallback) {
```

Comment:
I'm good with having a generic callback, but it's a bit confusing if the developer client using always the same generic param (`KinAccount`) and the other is for internal use.
Maybe we can have `RecoveryCallback` and share it between restore and backup?
or it's better to have precise callback as we don't have many options here for the generic value.